src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -560,6 +562,32 @@ public void startJenkins() throws Throwable {
             }
             Thread.sleep(100);
         }
+        addTimeout();
+    }
+
+    public static Optional<String> checkResult(HttpURLConnection conn) throws IOException {
```

Comment:
BTW I think a simple `@CheckForNull String` would suffice here. `Optional` is useful for certain advanced idioms (like `Stream`s) but just seems like overkill for direct use. Not sure if there is any evolving consensus on this stylistic point.